server/asset/eth/eth.go

```diff
@@ -189,7 +208,23 @@ func (eth *Backend) ValidateSecret(secret, contract []byte) bool {
 
 // Synced is true if the blockchain is ready for action.
 func (eth *Backend) Synced() (bool, error) {
-	return false, notImplementedErr
+	// node.SyncProgress will return nil both before syncing has begun and
+	// after it has finished. In order to discern when syncing has begun,
+	// ensure we are connected to at least requiredNPeers, assume the node
+	// has started syncing from those peers were they ahead, and then defer
+	// to syncProgress.
+	peers, err := eth.node.peers(eth.ctx)
+	if err != nil {
+		return false, err
+	}
+	if len(peers) < requiredNPeers {
+		return false, nil
+	}
```

Comment:
It's not clear to me that just having the peers is enough. In fact, I'm almost certain it's not. Looking through go-ethereum, I believe that `SyncProgress` maps to `(*PublicEthereumAPI).Syncing`. If `progress.CurrentBlock >= progress.HighestBlock`, you'll get the nil `*ethereum.SyncProgress`. `progress.HighestBlock` is set according to `Downloader.syncStatsChainHeight`, which is set in `(*Downloader).syncWithPeer`, which is ultimately invoked via a goroutine in `(*chainSyncer) startSync`. In short, we could have peers, but still have not started syncing, and will have a `nil` `*ethereum.SyncProgress`.

Reply:
@JoeGruffins Since you aren't getting responses in their Discord, and it seems like the `eth_syncing` method is fundamentally flawed, I don't think it's out of line to create an issue in their github repo describing the issue and how workarounds are also incomplete.

Reply:
Ok.

Reply:
https://github.com/ethereum/go-ethereum/issues/23099

Reply:
They are in no hurry to comment it seems. Will try again to get a goodish temp solution in the meantime...

Reply:
Let's not let perfect get in the way of this.  Recall that we didn't even have a sync status at one point and it wasn't the end of the world.  If eth returns a non-nil sync status OR peers==0, we can say it is not synchronized.  Otherwise (nil sync status and peers>0) is a fairly decent indicator that it's alright to proceed.  Despite being possible to have no peers and no a nil syncstatus object, my understanding is that's not a long-lived state.